engines/director/graphics.cpp

```diff
@@ -352,14 +352,18 @@ void inkDrawPixel(int x, int y, int src, void *data) {
 										(~bSrc | bFor) & (bSrc | bBak));
 			}
 		} else {
-			*dst = src;
+			// Find the inverse of the colour and match it back to the palette if required
+			byte rSrc, gSrc, bSrc;
+			wm->decomposeColor<T>(src, rSrc, gSrc, bSrc);
+
+			*dst = wm->findBestColor(~rSrc, ~gSrc, ~bSrc);
 		}
 		break;
 	case kInkTypeTransparent:
-		*dst = p->applyColor ? (src & p->foreColor) | (*dst & ~src) : (*dst & ~src);
+		*dst = p->applyColor ? (~src | p->backColor) & (*dst | src) : *dst | src;
```

Comment:
We have bunch of tests here: https://github.com/scummvm/director-tests/tree/master/D3-mac

You may consider checking them (and extending the test suite :) )